docs/specification.md

```diff
@@ -0,0 +1,169 @@
+    - Prefix with SyntaxTrait: For traits related to syntax highlighting.
+
+5. Constants
+
+    - Prefix with MAX_: For maximum values.
+    - Prefix with MIN_: For minimum values.
+    - Prefix with DEFAULT_: For default values.
+    - Prefix with ERROR_: For error codes or messages.
+
+6. Macros
+
+    - Prefix with macro_: For general-purpose macros.
+    - Prefix with log_: For logging-related macros.
+    - Prefix with assert_: For assertion-related macros.
+    - Prefix with debug_: For debugging-related macros.
+
+## Versioning ! 
+> using SemVer !   
+### Semantic Versioning (SemVer)
+
+- SemVer defines three components of a version number:
+
+    - MAJOR: Incremented for incompatible API changes.
+    - MINOR: Incremented for adding functionality in a backwards-compatible manner.
+    - PATCH: Incremented for backwards-compatible bug fixes.
+
+- Additionally, pre-release versions and build metadata can be appended to the version number:
+
+    - Pre-release: Denoted by a hyphen and a series of dot-separated identifiers (e.g., 1.0.0-alpha, 1.0.0-beta.2).
+    - Build metadata: Denoted by a plus sign and a series of dot-separated identifiers (e.g., 1.0.0+build.123).
+
+- When to Increment Versions
+
+    - MAJOR Version (e.g., 1.0.0 to 2.0.0):
+        - Increment when you make incompatible API changes.
+        - This indicates that users may need to update their code to use the new version.
+
+    - MINOR Version (e.g., 1.0.0 to 1.1.0):
+        - Increment when you add functionality in a backwards-compatible manner.
+        - This indicates new features that do not break existing functionality.
+
+    - PATCH Version (e.g., 1.0.0 to 1.0.1):
+        - Increment when you make backwards-compatible bug fixes.
+        - This indicates that the release includes fixes but no new features or breaking changes.
+
+- Setting Up Your Versioning System
+
+    - Initial Version (e.g., 0.0.1):
+        - Start with a version like 0.0.1 for initial development.
+        - Use 0.x.x versions for pre-release, unstable, or experimental features.
+
+    - Pre-release Versions (e.g., 1.0.0-alpha):
+        - Use pre-release versions to indicate that the software is not yet stable.
+        - Common pre-release identifiers include alpha, beta, rc (release candidate).
+
+    - Stable Release (e.g., 1.0.0):
+        - Increment to 1.0.0 when you consider the project stable and ready for production use.
+        - This indicates that the API is stable and that breaking changes will be minimized.
+
+    - Subsequent Releases:
+        - Increment the PATCH version for bug fixes.
+        - Increment the MINOR version for new features.
+        - Increment the MAJOR version for breaking changes.
+
+## Idioms and Code Practices !  
+### Coding Idioms ! 
+
+- TDD at the core : 
+  - Core Tests for a feature [X] must be written first ! 
+  - Push inital deltas to the remote origin without any initial business logic ! 
```

Comment:
I dont see the value in doing this over pushing code after it has been written, we have the kanban board so we are in theory aware of what people are working on at all times.

Reply:
ig yes , so when other check , can actually see what's going to be tested , and give their remarks regardless of the business logic !